app/src/main/java/ru/prometeydev/movie/FragmentMoviesList.kt

```diff
@@ -44,27 +47,46 @@ class FragmentMoviesList : Fragment() {
     override fun onStart() {
         super.onStart()
 
-        movies = MoviesDataSource().getMovies()
-        loadData()
+        scope.launch {
+            context?.let {
+                loadData(it)
+            }
+        }
     }
 
     override fun onDetach() {
         recycler = null
-        movies = null
 
         super.onDetach()
     }
 
-    private fun loadData() {
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        recycler?.let {
+            outState.putParcelable(BUNDLE_RECYCLER_LAYOUT, it.layoutManager?.onSaveInstanceState())
+        }
+    }
+
+    private suspend fun loadData(context: Context) = withContext(Dispatchers.Main) {
+        movies = loadMovies(context)
         (recycler?.adapter as? MoviesAdapter)?.apply {
-            movies?.let { bindMovies(it) }
+            bindMovies(movies)
+        }
+        recyclerRestoreState()
+    }
+
+    private fun recyclerRestoreState() {
+        arguments?.let {
+            val savedRecyclerLayoutState: Parcelable? = it.getParcelable(BUNDLE_RECYCLER_LAYOUT)
+            recycler?.layoutManager?.onRestoreInstanceState(savedRecyclerLayoutState)
         }
     }
 
     private fun doOnClick(movie: Movie) {
-        if (!movie.hasFullInfo()) {
-            view?.let { showMessageMissed(it) }
-            return
+        recycler?.let {
+            arguments = Bundle().apply {
+                putParcelable(BUNDLE_RECYCLER_LAYOUT, it.layoutManager?.onSaveInstanceState())
```

Comment:
Why on save instance state is not enough?

Reply:
Because onSaveInstanceState occurs when the screen is rotated, and when you go to the next screen by clicking on the card, it is not called.